cmd/node_agent/main.go

```diff
@@ -61,7 +61,23 @@ func main() {
 	}
 }
 
-func runNodeAgent() {
+func runNodeAgent(flags *flag.FlagSet) {
+	cc, err := platform.NewClientConfig(naConfig.Env)
+	if err != nil {
+		glog.Error(err)
+		os.Exit(-1)
+	}
+
+	// parse custom flags
+	cFlags := cc.GetFlagSet()
+	cFlags.AddFlagSet(flags)
+	err = cFlags.Parse(os.Args[1:])
```

Comment:
Please make sure you have tested this manually. Also, can you write unit test for this function?